jupyter_bokeh/widgets.py

```diff
@@ -126,13 +127,13 @@ def _sync_model(self, _, content, _buffers):
             submodel = self._model.select_one({"id": content["id"]})
             descriptor = submodel.lookup(content['attr'])
             try:
-                descriptor._real_set(submodel, old, new, hint=hint, setter=self)
+                descriptor._set(submodel, old, new, hint=hint, setter=self)
```

Comment:
If we could find a public API for this that would be ideal.

Reply:
Happy consider any suggestions that come with downstream tests.

Reply:
That said, I am also perfectly happy to keep it private and maintain it with tests and an understanding that `jupyter_bokeh` is part of Bokeh just maintained in a separate repo. Another overarching issue with Bokeh is that far too many things were made "public" API that are of no use  or interest to any normal users and I would very much like to reduce this surface.

Reply:
Adding: I had tried to clarify all this with the "General API" vs "Dev API" distinction, but that is just confusing for users and hard to maintain and curate, and very hard to document efectively. At this point, personally, I would like to work toward paring the public API down to only just what regular users  need to know about, and then work together on internally documenting and maintaining specific underscore APIs that are intended for cross-project use (e.g. maybe a naming convention `_internal_foo` for an internal maintained API and `_foo` for "truly private, not to rely on"